### PR TITLE
IFTAQCAA-14 - test script to cover mobile apps links

### PR DESCRIPTION
**Pull request by dkrutenko**
Click through mobile apps links to see if they work

### CODE REVIEW

**Comment by Nikodych** [comment]
looks like I would review two PR's, I would wait until we merge first one, because this includes code from the first

**Comment by Nikodych** [comment]
let me know, when it would be synced with master

**Comment by dkrutenko** [comment]
merged

File: src/main/java/com/softserveinc/ita/utils/WindowTabUtil.java
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/utils/WindowTabUtil.java:9]: BrowserTabHelper

File: src/main/java/com/softserveinc/ita/utils/WindowTabUtil.java
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/utils/WindowTabUtil.java:10]: empty line after class declaration

Comment (Nikodych) [reply src/main/java/com/softserveinc/ita/utils/WindowTabUtil.java:10]: all operations with selectors should be only in page objects or element util

such class should provide separate operations for the browser, take a look at Max's PR, calling methods for switching between tabs from class, and calling in test from page method for getting title

Comment (Nikodych) [reply src/main/java/com/softserveinc/ita/utils/WindowTabUtil.java:10]: create common page object for playmarkets with methods getAndroidAppTitle and ios

File: src/test/java/com/softserveinc/ita/HomePageTest.java
Comment (Nikodych) [inline src/test/java/com/softserveinc/ita/HomePageTest.java:71]: I thinks here not necessary to use data provider, just create variables in test

**Review by Nikodych — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/utils/WindowTabUtil.java
Comment (Nikodych) [reply src/main/java/com/softserveinc/ita/utils/WindowTabUtil.java:10]: Max's pr is merged

File: src/main/java/com/softserveinc/ita/pageobjects/BasePage.java
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/BasePage.java:94]: Clicked on header menu button

File: src/main/java/com/softserveinc/ita/pageobjects/HomePage.java
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/HomePage.java:54]: for get methods steps not neeeded

File: src/main/java/com/softserveinc/ita/pageobjects/ProductPage.java
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/ProductPage.java:12]: same

File: src/main/java/com/softserveinc/ita/pageobjects/SearchResultPage.java
Comment (Nikodych) [reply src/main/java/com/softserveinc/ita/pageobjects/SearchResultPage.java:19]: and for get and boolean methods steps not needed

File: src/main/java/com/softserveinc/ita/utils/BrowserTabHelper.java
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/utils/BrowserTabHelper.java:10]: add steps here

"Switched to second tab"

File: src/main/java/com/softserveinc/ita/utils/BrowserTabHelper.java
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/utils/BrowserTabHelper.java:14]: same

File: src/main/java/com/softserveinc/ita/pageobjects/HeaderMenuModal.java
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/HeaderMenuModal.java:10]: step not needed for get/boolean methods

File: src/main/java/com/softserveinc/ita/pageobjects/HeaderMenuModal.java
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/HeaderMenuModal.java:19]: same

File: src/test/java/com/softserveinc/ita/HomePageTest.java
Comment (Nikodych) [inline src/test/java/com/softserveinc/ita/HomePageTest.java]: implement such interface:
```
public interface IUiTextEnum {

    static <T extends Enum<T> & IUiTextEnum> T fromValue(String name, Class<T> enumType) {
        EnumSet<T> all = allOf(enumType);

        for (T aEnum : all) {
            if (aEnum.name().equalsIgnoreCase(replace(name, SPACE, "_"))) {
                return aEnum;
            }
        }
        throw new IllegalArgumentException("Unknown " + enumType.getSimpleName() + ": " + name);
    }
}
```

create method in enum SocialMedia after implementing interface above

```
public static SocialMedia getSocialMediaBy(@NonNull String name) {
        return fromValue(name, SocialMedia.class);
    }
```

Comment (Nikodych) [reply src/test/java/com/softserveinc/ita/HomePageTest.java]: and after this you will be able replace this "SocialMedia.valueOf(socialMediaName.toUpperCase())" by "getSocialMediaBy(socialMediaName)"

**Review by Nikodych — CHANGES_REQUESTED** [review]

File: src/test/java/com/softserveinc/ita/HomePageTest.java
Comment (dkrutenko) [reply src/test/java/com/softserveinc/ita/HomePageTest.java]: this method was in master already, social media links was made by Maks

Comment (Nikodych) [reply src/test/java/com/softserveinc/ita/HomePageTest.java]: ok, probably create task on this and add my comments to description

**Review by Nikodych — APPROVED** [review]

**Pull request merged**
